src/plugins/data_source_management/public/components/utils.test.ts

```diff
@@ -274,7 +278,26 @@ describe('DataSourceManagement: Utils.ts', () => {
       expect(getDefaultAuthMethod(authenticationMethodRegistery)?.name).toBe(AuthType.NoAuth);
     });
   });
+  describe('handleSetDefaultDatasourceAfterDeletion', () => {
+    test('should remove defaultDataSource setting and set new defaultDataSource if data sources exist', async () => {
+      mockResponseForSavedObjectsCalls(savedObjects.client, 'find', getDataSourcesResponse);
+      mockUiSettingsCalls(uiSettings, 'get', 'test');
 
+      await handleSetDefaultDatasourceAfterDeletion(savedObjects.client, uiSettings);
+      expect(uiSettings.set).toHaveBeenCalled();
+    });
+  });
+  describe('handleSetDefaultDatasourceDuringCreation', () => {
+    test('should set defaultDataSource if only one data source exists', async () => {
```

Comment:
For unit test of this function, I think it's worth to add a test to cover the case that when number of data source > 1 that `uiSettings.set` should not be called.

Reply:
Added!